lib/ansible/module_utils/facts/hardware/linux.py

```
@@ -0,0 +1,663 @@
+        cpu_facts['processor'] = []
+        for line in get_file_lines('/proc/cpuinfo'):
+            data = line.split(":", 1)
+            key = data[0].strip()
+
+            if xen:
+                if key == 'flags':
+                    # Check for vme cpu flag, Xen paravirt does not expose this.
+                    #   Need to detect Xen paravirt because it exposes cpuinfo
+                    #   differently than Xen HVM or KVM and causes reporting of
+                    #   only a single cpu core.
+                    if 'vme' not in data:
```

Ruff (0.12.2)

Use a single `if` statement instead of nested `if` statements

Combine `if` statements using `and`

(SIM102)